AdminClient/src/app/supplier/store/suppliers.effects.ts

```diff
@@ -0,0 +1,145 @@
+          tap(() => this.dialog.closeAll())
+      ))
+  ));
+
+  editSupplier$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.editSupplier),
+      switchMap((action: Action & { editedSupplier: SupplierModel }) => {
+        return of(this.editCustomer(action.editedSupplier)).pipe(delay(2000));
+      })
+  ));
+
+  showDeleteSupplierDialog$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.showDeleteSupplier),
+      map((action: Action & { focusedSupplier: SupplierModel }) => {
+        this.dialog.open<PopupDialogComponent, PopupDialogDataModel>(
+            PopupDialogComponent,
+            {
+              data: {
+                title: 'Delete supplier',
+                description: 'Are you sure you want to delete this supplier?',
+                dialogActions: [
+                  {
+                    text: 'Confirm',
+                    action: () => this.store$.dispatch(SupplierActions.deleteSupplier()),
+                    color: 'warn'
+                  },
+                  {
+                    text: 'Cancel',
+                    action: () => this.dialog.closeAll(),
+                    color: 'primary'
+                  }
+                ],
+                isDisabled: this.store$.select(fromSuppliers.selectIsLoading)
+              }
+            }
+        );
+      })
+  ), {dispatch: false});
+
+  showEditSupplierDialog$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.showEditSupplier),
+      map((action: Action & { focusedSupplier: SupplierModel }) => {
+        this.dialog.open<SupplierFormComponent, SupplierModel>(SupplierFormComponent,
+            {
+              width: '66vw',
+              height: '66vh',
+              panelClass: 'customerDialog'
+            }
+        );
+      })
+  ), {dispatch: false});
+
+  loadAllClients$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.beginLoadingSuppliers),
+      switchMap((action: Action) => of(this.loadAllClients()).pipe(delay(2000)))
+      )
+  );
+
+  constructor(
+      private dialog: MatDialog,
+      private actions$: Actions,
+      private store$: Store<fromApp.State>
+  ) {
+  }
+
+  createDummyClients() {
+    const dummySuppliers: SupplierModel[] = [];
+
+    for (let i = 0; i < 50; i++) {
+      dummySuppliers.push({
+        companyName: `Company ${ i }`,
+        contactNumber: `${ i.toString().repeat(9) }`,
+        description: `Some amazing company ${ i }`,
+        email: `bla${ i }@bla.com`,
+        firstName: `first name ${ i }`,
+        lastName: `last name ${ i }`,
+        id: `${ i }`,
+        notes: `look at me i am note ${ i } FEAR ME!`
+      });
+    }
+
+    return dummySuppliers;
+  }
+
+  loadAllClients(): Action {
+    const allSuppliers = this.createDummyClients();
+
+    return SupplierActions.allSuppliersLoaded({suppliers: allSuppliers});
+  };
+
+  private editCustomer(editedSupplier: SupplierModel): Action {
```

Comment:
```suggestion
  private editSupplier(editedSupplier: SupplierModel): Action {
```

And check references